[docs] remove confusing per-user controller reference (#6751)
skypilot-org/skypilot: docs/source/examples/managed-jobs.rst

How it works: The jobs controller
---------------------------------

> The jobs controller is a small on-demand CPU VM or pod running in the cloud that manages all jobs.

Reviewer: ```suggestion
The jobs controller is a small on-demand CPU VM or pod that manages all managed jobs for an API server. 
```
Is this clearer?
